provider-forums/src/java/org/sakaiproject/archiver/provider/ForumsArchiver.java

```diff
@@ -83,62 +84,189 @@ private void archiveForum(final DiscussionForum forum, final String archiveId, f
 		final SimpleForum simpleForum = new SimpleForum(forum);
 
 		// Initialise the array to hold each topic for this forum
-		final List<SimpleTopic> simpleTopics = new ArrayList<>();
+		List<SimpleTopic> simpleTopics = new ArrayList<>();
 
 		final List<DiscussionTopic> topics = forum.getTopics();
+
 		for (final DiscussionTopic topic : topics) {
-			final SimpleTopic simpleTopic = new SimpleTopic(topic);
 
-			// Add this topic to the forum's topic array
-			simpleTopics.add(simpleTopic);
+			simpleTopics = archiveTopics(forum, topic, simpleTopics, archiveId, siteId, includeStudentContent);
```

Comment:
Do topics need to be sorted or are they ok? I guess they are in date order already?

Reply:
Yep they are already fine